src/sentry/monitors/endpoints/monitor_ingest_checkin_index.py

```diff
@@ -210,6 +216,12 @@ def post(
                         return self.respond(status=200)
                     return self.respond(serialize(checkin, request.user), status=200)
             else:
+                if status == CheckInStatus.IN_PROGRESS:
+                    timeout_at = checkin.date_added.replace(second=0, microsecond=0) + timedelta(
+                        minutes=(monitor_config or {}).get("max_runtime") or TIMEOUT
+                    )
+                    checkin.update(timeout_at=timeout_at)
```

Comment:
We should probably figure out how to do this in the create instead, rather than adding an extra db mutation

Reply:
fair. I was trying to avoid manually creating the `date_added` but I can do that up above